sys/net/gnrc/network_layer/ndp/gnrc_ndp.c

```diff
@@ -260,6 +275,102 @@ void gnrc_ndp_nbr_adv_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
     return;
 }
 
+void gnrc_ndp_rtr_adv_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt, ipv6_hdr_t *ipv6,
+                             ndp_rtr_adv_t *rtr_adv, size_t icmpv6_size)
+{
+    uint8_t *buf = (uint8_t *)(rtr_adv + 1);
+    gnrc_ipv6_nc_t *nc_entry = NULL;
+    gnrc_ipv6_netif_t *if_entry = gnrc_ipv6_netif_get(iface);
+    uint8_t l2src[GNRC_IPV6_NC_L2_ADDR_MAX];
+    int sicmpv6_size = (int)icmpv6_size, l2src_len = 0;
+    uint16_t opt_offset = 0;
+    assert(if_entry != NULL);
+    if (!ipv6_addr_is_link_local(&ipv6->src) ||
+        ipv6_addr_is_multicast(&ipv6->src) ||
+        (ipv6->hl != 255) || (rtr_adv->code != 0) ||
+        (icmpv6_size < sizeof(ndp_rtr_adv_t))) {
+        DEBUG("ndp: router advertisement was invalid\n");
+        /* ipv6 releases */
+        return;
+    }
+    /* get source from default router list */
+    nc_entry = gnrc_ipv6_nc_get(iface, &ipv6->src);
+    if (nc_entry == NULL) { /* not in default router list */
+        /* create default router list entry */
+        nc_entry = gnrc_ipv6_nc_add(iface, &ipv6->src, NULL, 0,
+                                    GNRC_IPV6_NC_IS_ROUTER);
+        if (nc_entry == NULL) {
+            DEBUG("ndp: error on default router list entry creation\n");
+            return;
+        }
+    }
+    else if ((nc_entry->flags & GNRC_IPV6_NC_IS_ROUTER) && (byteorder_ntohs(rtr_adv->ltime) == 0)) {
+        nc_entry->flags &= ~GNRC_IPV6_NC_IS_ROUTER;
+    }
+    else {
+        nc_entry->flags |= GNRC_IPV6_NC_IS_ROUTER;
+    }
+    /* set router life timer */
+    if (rtr_adv->ltime.u16 != 0) {
+        vtimer_remove(&nc_entry->rtr_timeout);
+        vtimer_set_msg(&nc_entry->rtr_timeout,
+                       timex_set(byteorder_ntohs(rtr_adv->ltime), 0),
+                       thread_getpid(), GNRC_NDP_MSG_RTR_TIMEOUT, nc_entry);
+    }
+    /* set current hop limit from message if available */
+    if (rtr_adv->cur_hl != 0) {
+        if_entry->cur_hl = rtr_adv->cur_hl;
+    }
+    /* set flags from message */
+    if_entry->flags &= 0x3f;
+    if_entry->flags |= (rtr_adv->flags & (0xc0));
+    /* set reachable time from message if it is not the same as the random base
+     * value */
+    if ((rtr_adv->reach_time.u32 != 0) &&
+        (if_entry->reach_time_base != byteorder_ntohl(rtr_adv->reach_time))) {
+        /* calculate new random reachable time */
+        uint32_t reach_time = genrand_uint32_range(GNRC_NDP_MIN_RAND, GNRC_NDP_MAX_RAND);
+        if_entry->reach_time_base = byteorder_ntohl(rtr_adv->reach_time);
+        reach_time = (reach_time * if_entry->reach_time_base) / 10;
```

Comment:
Ok, so, how about a comment here saying something like "to avoid floating point number computations, the boundaries for the random values are multiplied by 10"?